packages/core/src/http-handler.ts

```diff
@@ -28,30 +28,18 @@ export class HttpHandler {
     this._routes.registerController(name, spec);
   }
 
-  protected _handleRequest(request: ServerRequest, response: ServerResponse): Promise<void> {
+  protected async _handleRequest(request: ServerRequest, response: ServerResponse): Promise<void> {
     const parsedRequest: ParsedRequest = parseRequestUrl(request);
     const requestContext = this._createRequestContext(request, response);
 
-    // TODO(bajtos) bind findRoute to requestContext
-    const findRoute: FindRoute = (req) => {
-      const found = this._routes.find(req);
-      if (!found) {
-        throw new HttpErrors.NotFound(
-          `Endpoint "${req.method} ${req.path}" not found.`);
-      }
-      this._bindRouteInfo(requestContext, found.controller, found.methodName);
-      return found;
-    };
-
-    // TODO(bajtos) bind invoke to requestContext
-    const invoke: InvokeMethod = async (controllerName, method, args) => {
-      const controller: { [opName: string]: Function } = await requestContext.get(controllerName);
-      const result = await controller[method](...args);
-      return result;
-    };
+    this._bindFindRoute(requestContext);
+    this._bindInvokeMethod(requestContext);
+
+    const findRoute = await requestContext.get('findRoute');
+    const invokeMethod = await requestContext.get('invokeMethod');
```

Comment:
> whether we should move all the above lines to the application level

Yes, eventually we should move them. 

However, we need these methods to be bound in the root (application-level) context to allow applications to customize it. I don't think our bindings are flexible enough to support that case, so let's leave it for the next iteration and land this pull request as it is now.